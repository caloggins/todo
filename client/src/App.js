import React, { Component } from 'react'
import { connect } from 'react-redux'
import TaskForm from './TaskForm'
import './App.css'

export class App extends Component {
    renderItems() {
        return this.props.tasks.map(task => {
            return (
                <div className="list-group-item" key={task.id}>
                    <div>{task.description}</div>
                    <div className="btn btn-default">Delete</div>
                </div>
            )
        })
    }

    render() {
        return (
            <div className="App">
                <div className="list-group">{this.renderItems()}</div>
                <TaskForm />
            </div>
        )
    }
}

function mapStateToProps({ tasks }) {
    return { tasks: tasks }
}

export default connect(mapStateToProps)(App)
