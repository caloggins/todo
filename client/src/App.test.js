import React from 'react'
import { shallow, mount } from 'enzyme'
import configureStore from 'redux-mock-store'
import { App } from './App'

describe('App tests...', () => {
    describe('component tests', () => {
        it('renders correctly', () => {
            const props = {
                tasks: []
            }
            const tree = shallow(<App {...props} />)
            expect(tree).toMatchSnapshot()
        })

        it('it renders correctly', () => {
            const props = {
                tasks: [
                    { id: 1, done: false, description: 'Get groceries' },
                    { id: 2, done: true, description: 'Get gas' },
                    { id: 3, done: false, description: 'Pay bills' }
                ]
            }
            const tree = shallow(<App {...props} />)
            expect(tree).toMatchSnapshot()
        })
    })
})
