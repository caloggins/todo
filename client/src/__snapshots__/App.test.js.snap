// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests... component tests it renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="list-group"
  >
    <div
      className="list-group-item"
      key="1"
    >
      <div>
        Get groceries
      </div>
      <div
        className="btn btn-default"
      >
        Delete
      </div>
    </div>
    <div
      className="list-group-item"
      key="2"
    >
      <div>
        Get gas
      </div>
      <div
        className="btn btn-default"
      >
        Delete
      </div>
    </div>
    <div
      className="list-group-item"
      key="3"
    >
      <div>
        Pay bills
      </div>
      <div
        className="btn btn-default"
      >
        Delete
      </div>
    </div>
  </div>
  <Connect(TaskForm) />
</div>
`;

exports[`App tests... component tests renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="list-group"
  />
  <Connect(TaskForm) />
</div>
`;
